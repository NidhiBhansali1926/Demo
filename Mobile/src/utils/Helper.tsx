import { NavigationProp } from '@react-navigation/native';
import React, { ReactElement } from 'react';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';

import { store } from '../store/index';

import { AuthParamsList } from '@typing/navigation';

export const renderScreen = (screen: ReactElement) => {
  const myScreen = create(<Provider store={store}>{screen}</Provider>);
  return myScreen;
};

export const getPropsWithNavigation = (props?: unknown, navigationPropExtension?: Partial<NavigationProp<AuthParamsList>>) => ({
  ...props,
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    replace: jest.fn(),
    dispatch: jest.fn(),
    getState: jest.fn(() => {}),
    isFocused: () => true,
    addListener: jest.fn().mockImplementation((event, callback) => {
      callback();
      return {
        remove: jest.fn(),
      };
    }),
    ...navigationPropExtension,
  },
});
