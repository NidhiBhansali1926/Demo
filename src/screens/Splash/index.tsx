import { useFonts } from 'expo-font';
import React, { useEffect } from 'react';
import { ActivityIndicator, View } from 'react-native';

import { BG_IMAGE } from '@backgrounds';

import { AppBackground } from '@components';
import { useTheme } from '@hooks';
import { Routes } from '@routes';
import { CustomFonts } from '@src/theme/assets/fonts';
import { LogoSVG } from '@svg';
import { SplashScreenProps } from '@typing/navigation';

const Splash = ({ navigation }: SplashScreenProps) => {
  const { Layout, Gutters, colorThemeVariable } = useTheme();
  const [isFontLoaded] = useFonts(CustomFonts);

  const init = async () => {
    await new Promise((resolve) =>
      setTimeout(() => {
        resolve(true);
      }, 2000),
    );
    navigation.replace(Routes.Auth);
  };

  useEffect(() => {
    if (isFontLoaded) {
      init();
    }
  }, [isFontLoaded]);

  return (
    <AppBackground colors={colorThemeVariable.takeSurveyGradient} overlayColor={colorThemeVariable.takeSurveyOverlay} source={BG_IMAGE.takeSurvey}>
      <View style={[Layout.fill, Layout.center]}>
        <LogoSVG />
        <ActivityIndicator size={'large'} style={[Gutters.largeVMargin]} color={colorThemeVariable.background} />
      </View>
    </AppBackground>
  );
};

export default Splash;
