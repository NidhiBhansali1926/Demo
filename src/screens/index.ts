export { default as SplashScreen } from './Splash';
export { default as TakeSurveyScreen } from './Auth/TakeSurvey';
export { default as ElevateAdventureScreen } from './Auth/ElevateAdventure';
export { default as BoostJourneyScreen } from './Auth/BoostJourney';
export { default as MaximizeExperienceScreen } from './Auth/MaximizeExperience';
export { default as EnterTheGloversScreen } from './Auth/EnterTheGlovers';
