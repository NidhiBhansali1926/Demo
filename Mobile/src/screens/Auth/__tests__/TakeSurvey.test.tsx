import React from 'react';
import { TakeSurveyScreen } from '@src/screens';
import { getPropsWithNavigation, renderScreen } from '@utils/Helper';

describe('Snapshot Testcase for Rendering TakeSurvey Screen Component', () => {
  test('TakeSurvey Screen Snapshot with Mocked Dispatch', () => {
    const dispatchMock = jest.fn();
    const useAppDispatchMock = jest.requireMock('@hooks').useAppDispatch;
    useAppDispatchMock.mockReturnValue(dispatchMock);
    const props = getPropsWithNavigation();
    const takeSurvey = renderScreen(<TakeSurveyScreen {...props} />).toJSON();
    expect(takeSurvey).toMatchSnapshot();
  });
});
