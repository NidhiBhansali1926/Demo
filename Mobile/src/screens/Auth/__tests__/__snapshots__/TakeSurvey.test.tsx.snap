// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testcase for Rendering TakeSurvey Screen Component TakeSurvey Screen Snapshot with Mocked Dispatch 1`] = `
<AppBackground
  colors={[]}
  overlayColor=""
>
  <View
    style={
      [
        {},
        {
          "marginTop": 77,
        },
      ]
    }
  >
    <LogoSVG
      style={{}}
    />
  </View>
  <View
    style={
      {
        "marginHorizontal": 20,
        "paddingBottom": 35,
      }
    }
  >
    <Text
      style={
        [
          undefined,
          undefined,
          {
            "marginBottom": 8,
          },
          {
            "color": undefined,
          },
        ]
      }
    >
      takeSurvey:title
    </Text>
    <Text
      style={
        [
          undefined,
          undefined,
          undefined,
          {
            "color": undefined,
          },
        ]
      }
    >
      takeSurvey:descriptionOne
    </Text>
    <Text
      style={
        [
          undefined,
          undefined,
          undefined,
          {
            "marginBottom": 4,
            "marginTop": 16,
          },
          {
            "color": undefined,
          },
        ]
      }
    >
      takeSurvey:descriptionTwo
    </Text>
    <AppButton
      handleClick={[Function]}
      isLoading={false}
      name="takeSurvey:primaryBtn"
    />
    <AppButton
      containerStyle={
        [
          {
            "marginVertical": 0,
          },
          {
            "backgroundColor": undefined,
          },
        ]
      }
      handleClick={[Function]}
      name="takeSurvey:secondaryBtn"
    />
  </View>
</AppBackground>
`;
